### ISSUE TITLE
Nova-deno format didn't honor the settings of Nova Editor

### ISSUE THREAD

**Issue by eggmud** [jaydenseric/nova-deno#7]
Nova-deno format didn't honor the settings of Nova Editor。

**Comment by jaydenseric** [comment]
To clarify, you mean these Nova settings for indentation, etc?

<img width="792" alt="Screen Shot 2021-09-29 at 11 28 24 am" src="https://user-images.githubusercontent.com/1754873/135188551-6a98a620-6792-46e2-a8e9-6154d6eabfb6.png">

Over the years I've come to the firm belief that all config and tools for formatting should be contained within the version controlled files and declared dependencies of the repo. This way everyone collaborating on the repo formats the files exactly the same (regardless of editor config), and the formatting can be enforced via CI such as GitHub Actions.

What needs to happen, is Nova editor should adopt the formatting settings of the repo, not the other way around. Since it's quite difficult for Nova to detect all the formatting tools that exist, and keep up with their options, for now the best thing is for the user to manually configure their Nova settings to be appropriate for the project they are working on.

Potentially it would be cool if Nova extensions could disable and override certain Nova formatting related settings while activated, but I'm not sure that's a feature team Nova would implement any time soon.

**Comment by eggmud** [comment]
yes, it seems that nova supports .editconfig, you can use individual .editconfig file in different projects.  I found the issue may be caused by deno lsp. I search the code repo of deno, but I didn't find the relavant code about textdocument/formatting request. and I manually changed the code of you plugin, hardcoded the insertSpaces and tabSize 4, but it was unlucky. The command of deno format  always formatted my code  using 2 spaces.
